src/content/en/2021/caching.md

```diff
@@ -17,4 +20,414 @@ featured_stat_3: TODO
+    </tr>
+    <tr>
+     <td><code class="no-wrap">public</code></td>
+     <td>Indicates that any cache can store the response, including the Browser and the CDN. This is assumed by default.</td>
+    </tr>
+    <tr>
+     <td><code class="no-wrap">no-cache</code></td>
+     <td>A cached resource must be revalidated before its use, via a conditional request, even if it is not marked as stale.</td>
+    </tr>
+    <tr>
+     <td><code class="no-wrap">must-revalidate</code></td>
+     <td>A stale cached entry must be revalidated before its use, via a conditional request.</td>
+    </tr>
+    <tr>
+     <td><code class="no-wrap">no-store</code></td>
+     <td>Indicates that the response must not be cached.</td>
+    </tr>
+    <tr>
+     <td><code class="no-wrap">private</code></td>
+     <td>The response is intended for a specific user and should not be stored by shared caches such as CDNs.</td>
+    </tr>
+    <tr>
+     <td><code class="no-wrap">immutable</code></td>
+     <td>Indicates that the cached entry will never change during its TTL and that revalidation is not necessary.</td>
+    </tr>
+  </table>
+  <figcaption>TODO: Caption the table above</figcaption>
+</figure>
+
+The `max-age` directive is the most commonly found since it directly defines the TTL the same way that the `Expires` header does.
+
+{{ figure_markup(
+  image="cache-control-directives.png",
+  caption="Usage of `Cache-Control` directives.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=1359015817&format=interactive",
+  sheets_gid="1944529311",
+  sql_file="cache_control_directives.sql"
+) }}
+
+Cache Control directives, 61% of mobile requests include a Cache-Control response header with a `max-age` directive.
+
+The most misused Cache Control directive is `set-cookie`, used for 0.07% of total directives for Desktop and 0.08% for Mobile.
+
+Fun fact the the largest max-age value is 1625540701883604800000 seconds or 51545557517871.8 years (51 trillion).
+
+{{ figure_markup(
+  caption="TODO",
+  content="51 trillion years",
+  classes="big-number",
+  sheets_gid="529870849",
+  sql_file="todo.sql"
+) }}
+
+Talk about `immutable` directive, 12% of desktop requests containing Cache-Control: immutable are on the "static.parastorage.com" host.
+
+### `304` Not Modified status
+
+{{ figure_markup(
+  image="http-304-by-caching-strategy.png",
+  caption="HTTP 304 response rate by caching strategy.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=1965033294&format=interactive",
+  sheets_gid="1136058277",
+  sql_file="valid_if_none_match_returns_304.sql"
+) }}
+
+### Validity of date strings
+
+Talk about invalid date, last modified and expires, (0.11% of dates in desktop requests are invalid).
+
```

Comment:
The three main HTTP headers used to represent timestamps, `Date`,`Last-Modified` and `Expires` all use a date formatted string. The `Date` HTTP response header is almost always generated automatically by the web server, meaning that invalid values are extremely rare. Still, in the rare event that the date is set incorrectly it can affect cacheability on the response on which it is served. 

CHART

Between 2020 and 2021, the percent using invalid date improved by a .5% but got worse for last-modified and expires showing that it was related to how the date was set on caching.